### ISSUE TITLE
Number of QuickCheck test cases is inconsistent

### ISSUE THREAD

**Issue by feuerbach** [nick8325/quickcheck#113]
@jtdaugherty reports that somewhere between QuickCheck  2.8.2 and 2.9.1 a bug (or a silent and undocumented in the Changelog change in semantics) was introduced: https://github.com/feuerbach/tasty/issues/147

Could you look into it?

**Comment by phadej** [comment]
that's related to
```
Remove "exhaustive" from Testable typeclass;
	  instead, combinators which are nonexhaustive (such as forAll)
	  call "again", which should be more robust
```

With
```diff
-    , testProperty "merged marking" $ property $ do
+    , testProperty "merged marking" $ property $ again $ do
```
I get
```
  merged marking:     OK (0.01s)
    +++ OK, passed 100 tests.
```

I'm not sure whether
```hs
 instance Testable prop => Testable (Gen prop) where
-  property mp = MkProperty $ do p <- mp; unProperty (property p)
+  property mp = MkProperty $ do p <- mp; unProperty (again p)
 ```
is the right solution on the `QuickCheck` side

**Comment by nick8325** [comment]
Thanks @phadej for the patch.

QuickCheck has a mechanism where the user can ask a property to be tested only once (useful e.g. when you just want to check that an `IO Bool` evaluates to true). In 2.9 I simplified the way this mechanism is implemented, with the side effect that a property will only be repeatedly tested if it uses `forAll` or similar. Unfortunately this was a bit too aggressive - @phadej's patch makes it so that the property will also be repeatedly tested if it uses the `Gen` monad.

**Comment by jtdaugherty** [comment]
Thanks! Can you mention here when this is released to Hackage?

**Comment by nick8325** [comment]
Should be fixed in 2.9.2 which is out now.

**Comment by jtdaugherty** [comment]
Thanks!
